backend/inspirehep/assign/views.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from flask import Blueprint, current_app, jsonify, request
+from inspire_dojson.utils import get_recid_from_ref, get_record_ref
+
+from inspirehep.disambiguation.utils import create_new_empty_author
+from inspirehep.records.api import AuthorsRecord, LiteratureRecord
+
+blueprint = Blueprint("inspirehep_assign", __name__, url_prefix="/assign")
+
+
+def get_literature_records_by_recid(recids):
+    pids = [("lit", str(recid)) for recid in recids]
+    return LiteratureRecord.get_records_by_pids(pids)
+
+
+def get_author_by_recid(literature_record, author_recid):
+    # can there be more than one author with same id? that's probably a bigger issue
+    return next(
+        author
+        for author in literature_record.get("authors")
+        if get_recid_from_ref(author.get("record")) == author_recid
+    )
+
+
+def unstub_author_by_recid(author_recid):
+    author = AuthorsRecord.get_record_by_pid_value(author_recid)
+    if author.get("stub") is True:
+        author["stub"] = False
+        author.update(dict(author))
+
+
+def assign_papers(from_author_recid, to_author_recid, literature_recids):
+    for record in get_literature_records_by_recid(literature_recids):
+        from_author = get_author_by_recid(record, from_author_recid)
+        from_author["record"] = get_record_ref(to_author_recid, endpoint="authors")
+        from_author["curated_relation"] = True
+        record.update(dict(record))
+
+
+def assign_to_new_stub_author(from_author_recid, literature_recids):
+    # do we need different data for this?, since it's not by BEARD
```

Comment:
Good point, indeed it would be better to have it with different data.

Reply:
but BEARD needs to be triggered or at least run for this author. Does this require any specific field to be present rather than `stub: true` ?